### PR TITLE
Mark some additional ucx-py tests for GPU

### PR DESCRIPTION
**Pull request by charlesbluca**
Noticed that there are some tests that require ucx-py that are currently not being run on gpuCI - added the pytest marker to them

cc @quasiben @jrbourbeau 

- [ ] Closes #xxxx
- [ ] Tests added / passed
- [ ] Passes `pre-commit run --all-files`

### CODE REVIEW

**Comment by fjetter** [comment]
`distributed.comm.tests.test_comms.test_ucx_client_server` promptly fails on gpuCI https://gpuci.gpuopenanalytics.com/job/dask/job/distributed/job/prb/job/distributed-prb/1077/CUDA_VER=11.2,LINUX_VER=ubuntu18.04,PYTHON_VER=3.8,RAPIDS_VER=22.02/testReport/junit/distributed.comm.tests/test_comms/test_ucx_client_server/

**Comment by charlesbluca** [comment]
rerun tests

**Review by jrbourbeau — COMMENTED** [review]
Thanks @charlesbluca. This change seems reasonable but, as @fjetter pointed out, `test_ucx_client_server` is failing. Any thoughts on what might be going on? Are you able to reproduce locally?

**Comment by charlesbluca** [comment]
I am able to reproduce locally, though I'm not too familiar with the UCX comms - @pentschev may be better for that. I do notice locally that despite the comm object being closed, the endpoint itself is still available:

```python
→ pytest distributed/comm/tests/test_comms.py::test_ucx_client_server --pdb
======================================== test session starts ========================================
platform linux -- Python 3.8.12, pytest-6.2.5, py-1.11.0, pluggy-1.0.0 -- /home/charlesb/miniconda3/envs/rapids-dask/bin/python
cachedir: .pytest_cache
benchmark: 3.4.1 (defaults: timer=time.perf_counter disable_gc=False min_rounds=5 min_time=0.000005 max_time=1.0 calibration_precision=10 warmup=False warmup_iterations=100000)
hypothesis profile 'default' -> database=DirectoryBasedExampleDatabase('/home/charlesb/dev/rapids-dask/distributed/.hypothesis/examples')
rootdir: /home/charlesb/dev/rapids-dask/distributed, configfile: setup.cfg
plugins: asyncio-0.12.0, forked-1.3.0, rerunfailures-10.2, timeout-2.0.2, cov-3.0.0, benchmark-3.4.1, xdist-2.5.0, anyio-3.4.0, repeat-0.8.0, hypothesis-6.34.1
timeout: 300.0s
timeout method: thread
timeout func_only: False
collected 1 item                                                                                    

distributed/comm/tests/test_comms.py::test_ucx_client_server FAILED                           [100%]
>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>> traceback >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>

    @pytest.mark.asyncio
    async def test_ucx_client_server():
        pytest.importorskip("distributed.comm.ucx")
        ucp = pytest.importorskip("ucp")
    
        addr = ucp.get_address()
>       await check_client_server("ucx://" + addr)

distributed/comm/tests/test_comms.py:546: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
distributed/comm/tests/test_comms.py:534: in check_client_server
    await asyncio.gather(*futures)
distributed/comm/tests/test_comms.py:525: in client_communicate
    msg = await comm.read()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <UCX (closed)  local= remote=ucx://172.25.122.215:34743>
deserializers = ('cuda', 'dask', 'pickle', 'error')

    async def read(self, deserializers=("cuda", "dask", "pickle", "error")):
        with log_errors():
            if self.closed():
>               raise CommClosedError("Endpoint is closed -- unable to read message")
E               distributed.comm.core.CommClosedError: Endpoint is closed -- unable to read message

distributed/comm/ucx.py:311: CommClosedError
>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>> entering PDB >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>

>>>>>>>>>>>>>>>>>>>>>>>>>>>>> PDB post_mortem (IO-capturing turned off) >>>>>>>>>>>>>>>>>>>>>>>>>>>>>
> /home/charlesb/dev/rapids-dask/distributed/distributed/comm/ucx.py(311)read()
-> raise CommClosedError("Endpoint is closed -- unable to read message")
(Pdb) self
<UCX (closed)  local= remote=ucx://172.25.122.215:34743>
(Pdb) self.closed()
True
(Pdb) self.ep
<ucp.core.Endpoint object at 0x7fb5db7fff40>
```

Which seems to conflict with the fact that closing the comm object should null the endpoint:

https://github.com/dask/distributed/blob/252b04572be5777516cf493f3b45417130ba6abb/distributed/comm/ucx.py#L369-L390

**Comment by pentschev** [comment]
> I am able to reproduce locally, though I'm not too familiar with the UCX comms - @pentschev may be better for that. I do notice locally that despite the comm object being closed, the endpoint itself is still available:

That's intentional. A UCX endpoint may still have incoming messages enqueued even after it closed, so sending messages isn't possible anymore, but receiving messages may be and we need to drain them all instead of really nullifying the endpoint right away.

To fix this I think we can get rid of the check for a closed endpoint at the top of the `UCX.read`, could you try removing the lines below?

https://github.com/dask/distributed/blob/252b04572be5777516cf493f3b45417130ba6abb/distributed/comm/ucx.py#L310-L312

**Comment by charlesbluca** [comment]
Thanks for the help @pentschev 🙂 tried that out and it resolved the failed test, I went ahead and pushed the change to check if this has an impact on any other tests

File: distributed/comm/ucx.py [hunk 1]
```diff
@@ -307,8 +307,8 @@ async def write(
 
     async def read(self, deserializers=("cuda", "dask", "pickle", "error")):
         with log_errors():
-            if self.closed():
-                raise CommClosedError("Endpoint is closed -- unable to read message")
+            # if self.closed():
+            #     raise CommClosedError("Endpoint is closed -- unable to read message")
 
             if deserializers is None:
```
Comment (pentschev) [inline distributed/comm/ucx.py:313]:
```suggestion
            if deserializers is None:
```

Comment (pentschev) [reply distributed/comm/ucx.py:313]: We can remove the check, no need to just comment it out.

**Review by pentschev — APPROVED** [review]
Just a minor code cleanup suggestion, but otherwise this LGTM since gpuCI passed. Thanks @charlesbluca !

**Review by jrbourbeau — APPROVED** [review]
Thanks @charlesbluca @pentschev -- test failures are unrelated

**Pull request merged**